src/golang/lib/workflow/utils/utils.go

```diff
@@ -88,6 +92,11 @@ func CleanupStorageFiles(ctx context.Context, storageConfig *shared.StorageConfi
 	}
 }
 
+func CheckIfObjectExistsInStorage(ctx context.Context, storageConfig *shared.StorageConfig, path string) bool {
+	_, err := storage.NewStorage(storageConfig).Get(ctx, path)
+	return err == nil
+}
```

Comment:
TODO: I still need to figure out the best way to check if the object exists here.